### PR TITLE
Support fish shell

### PR DESCRIPTION
**Pull request by lewisacidic**
Hi - thanks for the great library, such a helpful example of the power of treesitter.

I've implemented endwise queries for the [fish shell](https://fishshell.com) language. 

I also added some test cases, but I haven't actually been able to configure my system correctly in order to run them.  I think they each passed under manual testing...

Let me know if you need anything else from my end.

### CODE REVIEW

File: tests/endwise/fish.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+test "fish, global function", <<~END
+-function foo█
++function foo
++  
++end
+END
+
+test "fish, global function with options", <<~END
+-function foo --on-signal WINCH █
++function foo --on-signal WINCH
```
Comment (RRethy) [inline tests/endwise/fish.rb:18]:
```suggestion
+function foo --on-signal WINCH 
```

**Review by RRethy — CHANGES_REQUESTED** [review]
You can run the tests with `ruby ./tests/runner.rb fish`, although I should create a Makefile lol. I marked the failing test when I ran it locally.

File: tests/endwise/fish.rb (hunk 1 above)
Comment (lewisacidic) [reply tests/endwise/fish.rb:18]: Ooops I'll fix that.

**Comment by lewisacidic** [comment]
I think I tried that and had problems. Presumably my ruby is dodgy or neovim config is different or something. Would be awesome to set up CI with actions.

**Comment by lewisacidic** [comment]
Fix is in anyway :) Thanks for getting to it so fast!

**Pull request merged**